packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,646 @@
+
+    1. When a user inserts the character (e.g., `@`) into the editor, which activates the mention popup.
+    2. When a user adds or removes text after inserting the mention character into the editor.
+    3. When a user repositions the cursor between the text segments added after the mention character.
+    4. When there is configuration element `nimble-rich-text-mention-users` but with no mapping elements.
+
+    Refer the [accessibility](#accessibility) section to know more details about when it requires to emit the event for performing the filtering in the client application.
+
+    Conversely, this event will not be fired when:
+
+    1. A user moves the cursor away either using mouse or any key down events that moves the cursor from current position to outside of `@`
+       and characters after that with less than two white spaces while the popup is open.
+    2. A user selects an option from the dropdown list either using mouse click/pressing Enter or Tab.
+    3. A user adds two or more spaces after triggering the mention popup.
+    4. In the absence of `nimble-rich-text-mention-users` and no mapping elements.
+
+#### Other APIs for `@mention`
+
+To support the `@mention`, the creation of the following configuration elements and UI elements is required for rich text components:
+
+#### User mention element (Non-visible configuration element):
+
+This element serves as a container for the mapping elements and additionally functions as a configuration element. It allows the transmission of
+arbitrary values that are specific to the `@mention` users within the rich text components. Currently, this component does not have any
+configuration properties/attributes for user mentions and plays the role of containing element for the user mapping elements.
+
+This component can be used as a feature flag to enable the support for `@mention` in the rich text editor and viewer.
+If this component is not present at the children of the rich text editor or viewer, the component will function the same without the support
+for `@mention` in any place like editor, viewer, markdown parser and markdown serializer and the toolbar button to add **"@"** in the editor
+will be hidden.
+
+_Component Name_
+
+-   `nimble-rich-text-mention-users`
+
+_Content_
+
+-   Zero or more `nimble-mapping-mention-user` elements
+
+#### User mapping element (Non-visible configuration element):
+
+This mapping element is employed to establish a connection between the value displayed in the mapping view and the corresponding value stored within
+the markdown string. For instance, the username for an `@mention` is contained in the `display-name` attribute, which is used for display in the
+mention view, while the user ID URL is contained in the `url` attribute which is used to store within the markdown string.
+
+_Component Name_
+
+-   `nimble-mapping-mention-user`
+
+_Props/Attrs_
+
+-   `url`: string
```

Comment:
I'm not sure I'm a fan of the name `url` for this attribute, as it's a bit too generic. As this is still a value that is ultimately an id, would it be wrong to call is `url-id` (or some facsimile of that)?

Reply:
I think it can be named `mention-url` as it is specific to map the display name only for the mention feature. So I have updated the name in all places in the spec as suggested.